core/services/ocr2/plugins/threshold/decryption_queue_test.go

```diff
@@ -0,0 +1,357 @@
+
+	ctx3, cancel3 := context.WithTimeout(context.Background(), time.Duration(1000)*time.Millisecond)
+	defer cancel3()
+
+	//nolint:errcheck
+	go dq.Decrypt(ctx3, []byte("13"), []byte("encrypted"))
+
+	waitForPendingRequestToBeAdded(t, dq, []byte("13"))
+
+	requests := dq.GetRequests(2, 1000)
+	expected := []DecryptionRequest{
+		{[]byte("11"), []byte("encrypted")},
+		{[]byte("12"), []byte("encrypted")},
+	}
+	if !reflect.DeepEqual(requests, expected) {
+		t.Error("GetRequests did not return the expected requests when requestCountlimit is reached")
+	}
+}
+
+func TestGetRequestsBytesLimit(t *testing.T) {
+	dq := NewThresholdDecryptor(4, 10, 1000)
+
+	ctx1, cancel1 := context.WithTimeout(context.Background(), time.Duration(1000)*time.Millisecond)
+	defer cancel1()
+
+	//nolint:errcheck
+	go dq.Decrypt(ctx1, []byte("11"), []byte("encrypted"))
+
+	waitForPendingRequestToBeAdded(t, dq, []byte("11"))
+
+	ctx2, cancel2 := context.WithTimeout(context.Background(), time.Duration(1000)*time.Millisecond)
+	defer cancel2()
+
+	//nolint:errcheck
+	go dq.Decrypt(ctx2, []byte("12"), []byte("encrypted"))
+
+	waitForPendingRequestToBeAdded(t, dq, []byte("12"))
+
+	ctx3, cancel3 := context.WithTimeout(context.Background(), time.Duration(1000)*time.Millisecond)
+	defer cancel3()
+
+	//nolint:errcheck
+	go dq.Decrypt(ctx3, []byte("13"), []byte("encrypted"))
+
+	waitForPendingRequestToBeAdded(t, dq, []byte("13"))
+
+	requests := dq.GetRequests(4, 30)
+	expected := []DecryptionRequest{
+		{[]byte("11"), []byte("encrypted")},
+		{[]byte("12"), []byte("encrypted")},
+	}
+	if !reflect.DeepEqual(requests, expected) {
+		t.Error("GetRequests did not return the expected requests when totalBytesLimit is exceeded")
+	}
+}
+
+func TestGetRequestsWithShortPendingRequestQueue(t *testing.T) {
+	dq := NewThresholdDecryptor(4, 1000, 1000)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(1000)*time.Millisecond)
+	defer cancel()
+
+	//nolint:errcheck
+	go dq.Decrypt(ctx, []byte("11"), []byte("encrypted"))
+
+	waitForPendingRequestToBeAdded(t, dq, []byte("11"))
+
+	requests := dq.GetRequests(2, 1000)
+	expected := []DecryptionRequest{
+		{[]byte("11"), []byte("encrypted")},
+	}
+	if !reflect.DeepEqual(requests, expected) {
+		t.Error("GetRequests did not return the expected requests when limit is set")
+	}
+}
+
+func TestGetRequestsWithExpiredRequest(t *testing.T) {
+	dq := NewThresholdDecryptor(4, 1000, 1000)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(100)*time.Millisecond)
+	defer cancel()
+
+	//nolint:errcheck
+	dq.Decrypt(ctx, []byte("11"), []byte("encrypted"))
+
+	requests := dq.GetRequests(2, 1000)
+	expected := []DecryptionRequest{}
+	if !reflect.DeepEqual(requests, expected) {
+		t.Error("GetRequests did not return the expected requests when limit is set")
+	}
+}
+
+func TestStart(t *testing.T) {
+	dq := NewThresholdDecryptor(4, 1000, 1000)
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Never use `context.Background()` in tests. This will result in flakey/stale tests.
Please use `testutils.Context(t)`. This will bound the specific test lifetime to the global tests execution timing.